source/agora/consensus/protocol/Nominator.d

```diff
@@ -353,15 +353,28 @@ extern(D):
                     fail_reason, data);
                 return ValidationLevel.kInvalidValue;
             }
+
+            if (data.enrolls.length >= Enrollment.MinValidatorCount)
```

Comment:
There is already `validateConsensusData()`, these checks belong there.

Reply:
I didn't know the height of the block there. `uint64_t slot_idx`

Reply:
```D
auto slot_idx = this.ledger.getBlockHeight() + 1;
```

This is already inside of `validateConsensusData`:

```D
const ulong expect_height = this.getBlockHeight() + 1;
```

Reply:
If the value is from another node, I don't think it should be compared to the block height of the current node.

Reply:
If `slot_idx != this.ledger.getBlockHeight() + 1;` then we should return `ValidationLevel.kInvalidValue`. We cannot proceed with nominating or voting on a value unless we're voting for the next block height.

Reply:
I will change this part to your opinion, please check it next time.